slingshotapi/deploy.py

```
@@ -32,8 +34,54 @@ def deploy(cls, deployment_data):
+                region_deploy['version'] = deployment_data['version']
+                ids[endpoint] = cls._trigger_deployment(engines[endpoint], region_deploy)
+        return ids
+        # TODO update the version record with Slingshot workflow details
+
+    @classmethod
+    def _process_deploys(cls, deployment):
+        regex = re.compile('(\$\$META\.\w+)')
+        deployments = {}
+        regions = deployment.get("regions", {})
+        for region in regions:
+             deployments[region] = []
+        for step in deployment.get("pipeline"):
+            parameters = step.get("parameters", {})
+            params_string = json.dumps(parameters)
+            for region, params in regions.items():
+                region_string = params_string
+                region_string = region_string.replace('$$REGION', region)
+                to_be_replaced = regex.findall(region_string)
+                for one_item in to_be_replaced:
+                    param_name = one_item.strip('$$META.')
+                    try:
+                        param = cls._process_parameter(params[param_name])
+                        region_string = region_string.replace('"{}"'.format(one_item), str(param))
+                    except:
+                        raise cls.ParameterNotProvidedError("You failed to provide a value for the custom "
+                                                                "parameter {}". format(param_name))
+                new_step = deepcopy(step)
+                region_string = json.loads(region_string)
+                new_step["parameters"] = region_string
+                deployments[region].append(new_step)
+        return deployments
+
+    @classmethod
+    def _trigger_deployment(cls, engine_endpoint, region_deploy):
+        try:
+            workflow = requests.post("{}/workflow?stream_logs=false".format(engine_endpoint), json=region_deploy)
+            return workflow.text
+        except:
+            return workflow.text
```

We should return the exception as a string.